modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/index/CompanyIndexFactoryHelper.java

```diff
@@ -311,15 +313,14 @@ private void _processContributions(
 
 		indexConfigurationContributor.contributeSettings(settingsBuilder::put);
 
-		Settings settings = settingsBuilder.build();
-
-		if (!settings.isEmpty()) {
-			IndicesClient indicesClient = restHighLevelClient.indices();
+		IndicesClient indicesClient = restHighLevelClient.indices();
```

Comment:
i think we can pull `Settings settings = settingsBuilder.build();` outside of the forEachCompanyId loop and put it up here with indicesClient, so that we dont call build() multiple times unnecessarily.

we can also do the same for `Validator.isNull(_elasticsearchConfigurationWrapper.overrideTypeMappings())`, just assign it to a boolean ie `contributeMappings` and pass that into the loop

Reply:
actually you can just take a look at the 2 commits i pushed here https://github.com/BryanEngler/liferay-portal/commits/2024_05_29_pr1358_liferay_search_2024_05_22_lpd_8356_lpd_8357_lpd_8358_lpd_8359/